### PR TITLE
Client: small improvements before first minor release

### PR DESCRIPTION
**Pull request by IceKhan13**
### Summary

Client: small improvements before first minor release

### CODE REVIEW

File: client/purplecaffeine/core.py [hunk 1]
```diff
@@ -188,12 +198,11 @@ def export_to_shared_file(self, path) -> str:
         Returns:
             Full path of the file
         """
-        with open(
-            os.path.join(path, self.name + ".json"), "w", encoding="utf-8"
-        ) as trial_file:
+        filename = os.path.join(path, f"{self.uuid}.json")
```
Comment (mickahell) [inline client/purplecaffeine/core.py:201]: The goal here was to be able to share file with the name of the experiment so you can send a file with a good name to anyone

File: client/purplecaffeine/core.py [hunk 2]
```diff
@@ -343,10 +389,8 @@ def save(self, trial: Trial) -> str:
         Returns:
             self.path: path of the trial file
         """
-        trial_id = trial.name + datetime.now().strftime("%Y%m%d%H")
-        with open(
-            os.path.join(self.path, trial_id + ".json"), "w", encoding="utf-8"
-        ) as trial_file:
+        save_path = os.path.join(self.path, f"{trial.uuid}.json")
```
Comment (mickahell) [inline client/purplecaffeine/core.py:389]: In local, was more logical to me to have file we can understand what they are by their name

File: client/purplecaffeine/core.py [hunk 3]
```diff
@@ -245,6 +254,34 @@ def get(self, trial_id: str) -> Trial:
...
+            >>>     password="123"
+            >>> )
+
+        Args:
+            username: username
+            password: password
+            host: host of api server
+        """
+        self.username = username
+        self.host = host
```
Comment (mickahell) [inline client/purplecaffeine/core.py:273]: Host is also in the Configuration file :
https://github.com/IceKhan13/purplecaffeine/blob/f7bdf361b5c20f90a0af2562ff1cd138581d8cfa/client/purplecaffeine/helpers/conf.py#L10

**Review by mickahell — APPROVED** [review]
Not really suggestion but more commentary.

It has some stuff we aren't on the page I think ^^'

File: client/purplecaffeine/core.py (hunk 3 above)
Comment (IceKhan13) [reply client/purplecaffeine/core.py:273]: we will not use configuration files as each deployment will have their host in different place

File: client/purplecaffeine/core.py (hunk 2 above)
Comment (IceKhan13) [reply client/purplecaffeine/core.py:389]: good point. Maybe we can move files in this structure `<path>/<experiment_name>/<uuid>.json`

File: client/purplecaffeine/core.py (hunk 3 above)
Comment (mickahell) [reply client/purplecaffeine/core.py:273]: We should allow the people to select host they want and if they don't set anything just use the one from the Configuration

File: client/purplecaffeine/core.py (hunk 2 above)
Comment (IceKhan13) [reply client/purplecaffeine/core.py:389]: ah, then to load it back we will need to provide name and uuid =(

Comment (mickahell) [reply client/purplecaffeine/core.py:389]: Yeah, we don't have simple answer here :/

File: client/purplecaffeine/core.py (hunk 3 above)
Comment (mickahell) [reply client/purplecaffeine/core.py:273]: @IceKhan13 what do you mean they all gonna be different ?
We will always have only 1 simple url to call the api. The main DNS, not gonna change, or am I missing something ?

**Pull request merged**